docs/app/components/content/ComponentThemeVisualizer.vue

```diff
@@ -0,0 +1,168 @@
+
+  highlightedSlot.value = slotName
+
+  const containerRect = props.container.getBoundingClientRect()
+  const targetRect = slotElement.getBoundingClientRect()
+
+  const left = targetRect.left - containerRect.left
+  const top = targetRect.top - containerRect.top
+
+  animate.value = {
+    x: left,
+    y: top,
+    width: targetRect.width,
+    height: targetRect.height,
+    opacity: 1
+  }
+}
+
+function clearHighlight() {
+  highlightedSlot.value = null
+  animate.value = { opacity: 0 }
+}
+
+function isSlotRendered(slotName: string): boolean {
+  if (!props.container) return false
+
+  const elements = props.container.querySelectorAll(`[data-slot="${slotName}"]`)
+  for (const el of elements) {
+    if (!visualizerRef.value?.contains(el)) {
+      return true
+    }
+  }
+  return false
+}
+
+// Clear highlight when popover closes
+watch(open, (isOpen) => {
+  if (!isOpen) {
+    clearHighlight()
+  }
+})
+</script>
+
+<template>
+  <div ref="visualizerRef" class="group">
+    <UPopover
+      v-model:open="open"
+      :content="{ align: 'start' }"
+      :ui="{ content: 'w-64 max-h-72 overflow-y-auto' }"
+    >
+      <UButton
+        icon="i-lucide-scan-eye"
+        color="neutral"
+        variant="outline"
+        size="sm"
+        class="absolute -top-[11px] -right-[11px] z-55 rounded-full lg:opacity-0 lg:group-hover/component:opacity-100 ring-muted"
+        :class="[open && 'lg:opacity-100 bg-elevated']"
+      />
+
+      <template #content>
+        <div class="px-2.5 py-1.5 text-xs font-semibold text-highlighted border-b border-default">
+          Theme slots
+        </div>
+        <div class="p-1">
+          <div
+            v-for="slotName in themeSlots"
+            :key="slotName"
+            class="p-1.5 cursor-default hover:bg-elevated/50 transition-colors rounded"
+            :class="[highlightedSlot === slotName && 'bg-elevated/50']"
+            @mouseenter="highlightSlot(slotName)"
+            @mouseleave="clearHighlight"
+          >
+            <div class="flex items-center gap-2">
+              <code class="text-xs font-medium" :class="[isSlotRendered(slotName) ? 'text-highlighted' : 'text-muted']">{{ slotName }}</code>
+              <span v-if="!isSlotRendered(slotName)" class="text-[10px] text-muted">(not rendered)</span>
+            </div>
+            <div v-if="getSlotClasses(slotName)" class="mt-0.5 text-[10px] text-muted line-clamp-2 font-mono">
+              {{ getSlotClasses(slotName) }}
+            </div>
+          </div>
+          <div v-if="!themeSlots.length" class="p-1.5 text-xs text-muted">
+            No slots found
+          </div>
+        </div>
+      </template>
+    </UPopover>
+
+    <!-- Highlight overlay with motion animation -->
+    <component
+      :is="motion.div"
+      :animate="animate"
+      :transition="{ type: 'spring', stiffness: 500, damping: 30 }"
+      class="absolute top-0 left-0 pointer-events-none border border-dashed border-primary invert z-55"
```

Comment:
I think we can increase the border a bit to make it stand out more, and maybe add some border radius? 

```suggestion
      class="absolute top-0 left-0 pointer-events-none border-2 border-dashed border-primary invert z-55"
```